### PR TITLE
add feature: support config map source,

### PR DESCRIPTION
**Pull request by luckyyuyong**

### CODE REVIEW

File: sources/configmap-source/configmapsource.go [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+	cmSource.fileHandlers[path] = handle
+	fileType := fileType(fs)
+	switch fileType {
+	case Directory:
+		err := cmSource.addDirRecursion(path, priority, handle)
+		if err != nil {
+			openlogging.GetLogger().Errorf("Failed to handle directory [%s] %s", path, err)
+			return err
+		}
+	case RegularFile:
```
Comment (tianxiaoliang) [inline sources/configmap-source/configmapsource.go]: delete it

File: sources/configmap-source/configmapsource.go [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+	}
+
+	if cmSource.watchPool != nil {
+		cmSource.watchPool.AddWatchFile(path)
+	}
+
+	return nil
+}
+
+func (cmSource *configMapSource) addDirRecursion(path string, priority uint32, handle FileHandler) error {
```
Comment (tianxiaoliang) [inline sources/configmap-source/configmapsource.go]: use standard go lib to walk file folder

Comment (luckyyuyong) [reply sources/configmap-source/configmapsource.go]: fix it

File: sources/configmap-source/configmapsource.go (hunk 1 above)
Comment (luckyyuyong) [reply sources/configmap-source/configmapsource.go]: ok

**Comment by luckyyuyong** [comment]
please review

File: sources/utils/file_handler.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	"path/filepath"
+)
+
+//FileHandler decide how to convert a file content into key values
+//archaius will manage file content as those key values
+type FileHandler func(filePath string, content []byte) (map[string]interface{}, error)
+
+//Convert2JavaProps is a FileHandler
+//it convert the yaml content into java props
+func Convert2JavaProps(p string, content []byte) (map[string]interface{}, error) {
```
Comment (tianxiaoliang) [inline sources/utils/file_handler.go]: since you moves code here ,you should delete same code in file source package and change the import packge

**Pull request merged**